packages/react-devtools-shell/webpack-server.js

```diff
@@ -200,4 +182,48 @@ const e2eRegressionApp = semver.lt(REACT_VERSION, '18.0.0')
       },
     );
 
-module.exports = [app, e2eRegressionApp];
+const appCompiler = Webpack(app);
+const appServer = new WebpackDevServer(
+  {
+    hot: true,
+    open: true,
+    port: 8080,
+    client: {
+      logging: 'warn',
+    },
+    static: {
+      directory: __dirname,
+      publicPath: '/',
+    },
+  },
+  appCompiler,
+);
+
+const e2eRegressionAppCompiler = Webpack(e2eRegressionApp);
+const e2eRegressionAppServer = new WebpackDevServer(
+  {
+    port: 8181,
+    client: {
+      logging: 'warn',
+    },
+    static: [
+      {
+        publicPath: '/dist/',
+      },
+    ],
+  },
+  e2eRegressionAppCompiler,
+);
+
+const runServer = async () => {
+  console.log('Starting server...');
+
+  appServer.compiler.hooks.done.tap('done', () =>
+    console.log(SUCCESSFUL_COMPILATION_MESSAGE),
+  );
```

Comment:
This is the easiest way that I found on how to emit something (like an event) when webpack has compiled assets.

More scalable solution for it would be wrapping up in a Promise, but this server is being started via spawning node process in `scripts/circleci/run_devtools_e2e_tests.js`, so we need something like this ¯\\\_(ツ)\_/¯